### PR TITLE
[CI] Fix macOS release workflow

### PR DESCRIPTION
**Pull request by flavioislima**
After trying several things I figure it out that it would be a lot simpler just to use a npm lib that edits the YAML files and merge them together the way we need.
That should do the trick.

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [ ] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: .github/workflows/release_macOS.yml [hunk 1]
```diff
@@ -51,6 +51,15 @@ jobs:
         run: yarn release:mac:x64
         env:
           NOTARIZE: true
+      - name: Copy dist/latest-mac.yml to dist/latest-mac-x64.yml
+        run: |
+          cp dist/latest-mac.yml dist/latest-mac-x64.yml
+        shell: bash
+      - name: Save latest-mac-x64.yml as an artifact
+        uses: actions/upload-artifact@v2
```
Comment (BrettCleary) [inline .github/workflows/release_macOS.yml:59]: there is a v4 version of this action https://github.com/actions/upload-artifact?tab=readme-ov-file#v4---whats-new

File: .github/workflows/release_macOS.yml [hunk 2]
```diff
@@ -86,3 +95,64 @@ jobs:
         run: yarn release:mac:arm64
         env:
           NOTARIZE: true
+      - name: Copy dist/latest-mac.yml to dist/latest-mac-arm64.yml
+        run: |
+          cp dist/latest-mac.yml dist/latest-mac-arm64.yml
+        shell: bash
+      - name: Save latest-mac-arm64.yml as an artifact
+        uses: actions/upload-artifact@v2
```
Comment (BrettCleary) [inline .github/workflows/release_macOS.yml:103]: there is a v4 version of this action https://github.com/actions/upload-artifact?tab=readme-ov-file#v4---whats-new

File: .github/workflows/release_macOS.yml [hunk 3]
```diff
@@ -86,3 +95,64 @@ jobs:
...
+
+  generate-latest-mac:
+    needs:
+      - release-x64
+      - release-arm64
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
```
Comment (BrettCleary) [inline .github/workflows/release_macOS.yml:116]: v4

Comment (BrettCleary) [reply .github/workflows/release_macOS.yml:116]: https://github.com/actions/checkout?tab=readme-ov-file#checkout-v4

File: .github/workflows/release_macOS.yml [hunk 4]
```diff
@@ -86,3 +95,64 @@ jobs:
...
+          ssh://git@github.com/
+
+      - name: Authenticate with private NPM package
+        run: echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" > ~/.npmrc
+
+      - name: Install dependencies
+        run: yarn setup
+
+      - name: Download artifacts from release-x64 and release-arm64 jobs
+        uses: actions/download-artifact@v2
```
Comment (BrettCleary) [inline .github/workflows/release_macOS.yml:136]: v4 https://github.com/actions/download-artifact?tab=readme-ov-file#v4---whats-new

File: .github/workflows/release_macOS.yml [hunk 5]
```diff
@@ -86,3 +95,64 @@ jobs:
...
+        with:
+          name: latest-mac-x64
+          path: dist-x64
+
+      - uses: actions/download-artifact@v2
```
Comment (BrettCleary) [inline .github/workflows/release_macOS.yml:141]: v4

File: .github/workflows/release_macOS.yml [hunk 6]
```diff
@@ -86,3 +95,64 @@ jobs:
...
+        with:
+          name: latest-mac-arm64
+          path: dist-arm64
+
+      - name: Merge YMLs
+        run: |
+          npm install js-yaml
```
Comment (BrettCleary) [inline .github/workflows/release_macOS.yml:148]: curious why we did npm instead of yarn here

File: merge-yml.js
Comment (BrettCleary) [inline merge-yml.js:1]: niceee

**Review by BrettCleary — APPROVED** [review]
🔥🔥🔥

File: .github/workflows/release_macOS.yml (hunk 6 above)
Comment (flavioislima) [reply .github/workflows/release_macOS.yml:148]: copy paste from chatgpt, changed just a few things and left this one here.
I will do some cleanup and update the manifests later

**Comment by BrettCleary** [comment]
> @BrettCleary will address your comments later, but at least it worked fine now! 🎉

sweet! yup, happy to handle on different PR

**Comment by flavioislima** [comment]
> > @BrettCleary will address your comments later, but at least it worked fine now! 🎉
> 
> sweet! yup, happy to handle on different PR

yeah, I am up for a PR to update all ymls with new packages and node 20 💪🏽

**Comment by flavioislima** [comment]
Decided to fix everything from your comments on this PR itself @BrettCleary 
Tested again and worked fine:

```version: 0.11.4
files:
  - url: HyperPlay-0.11.4-macOS-x64.zip
    sha512: dYVjMdNfmWz0UOd74DzX89J4ymrVsYvQgT5pge1yLkNkbBkRASXnLtCVtiTk4n9pbYP8ugyzhtq9j9XK+Q2Wzg==
    size: 234474277
  - url: HyperPlay-0.11.4-macOS-x64.dmg
    sha512: QHTS9eMOr/edj55Me9Iin3Egj/Xk/XALevZsovllLLcFOvxlVTOXS9SeOUJhGnJ7wVYoE3Xp7KzPVtGImYsopA==
    size: 243682703
  - url: HyperPlay-0.11.4-macOS-arm64.zip
    sha512: KH1ei0pv+g3cTAZflun2zwn7T8opuauKqZ1FxY5hlmTzL7qI1uJ7ThvQLhIcDWafye6oRv+jDsPd3pOdpLaULw==
    size: 226475636
  - url: HyperPlay-0.11.4-macOS-arm64.dmg
    sha512: yHpnYCzrYwDYsnnz4ljd8lkG4xJw1d2PI13fva+ak5wjjbp0U5PcbIVhZ9NaLo5SqCTC1kEm6xTnNoeX0f4Jbw==
    size: 235811402
path: HyperPlay-0.11.4-macOS-x64.zip
sha512: dYVjMdNfmWz0UOd74DzX89J4ymrVsYvQgT5pge1yLkNkbBkRASXnLtCVtiTk4n9pbYP8ugyzhtq9j9XK+Q2Wzg==
releaseDate: '2024-02-26T12:10:12.536Z'
```

**Pull request merged**